### PR TITLE
feat: Support constructors in PHP SDK

### LINKED ISSUE #8089: PHP: Add module constructor to runtime
### What are you trying to do?

Add support for passing constructor arguments to PHP SDK modules.

### Why is this important to you?

It would allow additional arguments to be defined that may be important to multiple functions on the same module.

i.e. The every function in the PhpSdkDev module requires defining a `source` argument. 
If instead we could rely on a constructor, it would only need to be defined once. Saving time and loc

### How are you currently working around this?

Repetitively defining the same argument in multiple functions

### PR DESCRIPTION
**Pull request by charjr**
close #8089

### CODE REVIEW

File: sdk/php/src/ValueObject/DaggerFunction.php [hunk 1]
```diff
@@ -19,6 +19,11 @@ public function __construct(
     ) {
     }
 
+    public function isConstructor(): bool
+    {
+        return $this->name === '__construct';
```
Comment (helderco) [inline sdk/php/src/ValueObject/DaggerFunction.php:24]: The name of the function registered as the constructor has to be an empty string (`""`).

**Comment by TomChv** [comment]
Giving an early approval, please rebase the PR first.

File: sdk/php/runtime/template/README.md [hunk 2]
```diff
@@ -173,10 +173,67 @@ class Example
 }
 ```
 
+#### Constructor
+
+You can expose PHP's magic method `__construct` as a `DaggerFunction`.
+If you do this, Dagger will use it as a constructor.
+
+This is only applicable to the base class of your Module: the class named identically to your Module.
```
[the remark this answers is not in the document]
Comment (jedevc) [reply sdk/php/runtime/template/README.md:181]: :eyes: this is the case for all constructors at the dagger api level.

**Comment by charjr** [comment]
> Giving an early approval, please rebase the PR first.
:+1: 

> I think it's also worth it to add some tests in the integration suites for that [here](https://github.com/dagger/dagger/blob/907c013a020eb9801af7166a84e5076719efb361/core/integration/module_php_test.go#L13)
> 
> You can see some examples [there](https://github.com/dagger/dagger/blob/7fb1fb85f72c7ddedff5807825aa311ac6216516/core/integration/module_test.go#L1180).

Thank you for linking to the examples, that is incredibly helpful!
I will make sure to do so, I will do so in a separate PR so I can keep them small.

File: sdk/php/runtime/template/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (helderco) [reply sdk/php/runtime/template/README.md:181]: It's true, the only constructor that can be registered is the one from the main object. For all other objects, their constructors are the object functions that chain them.

It doesn't mean custom object constructors aren't used. They can be used inside functions, but only within the language. They're not registered with Dagger as constructors.

**Pull request merged**